pwiz_tools/Skyline/Model/Results/SpectrumFilter.cs

```diff
@@ -173,15 +173,15 @@ public SpectrumFilter(SrmDocument document, MsDataFileUri msDataFileUri, IFilter
                     _isHighAccProductFilter = !Equals(_fullScan.ProductMassAnalyzer,
                         FullScanMassAnalyzerType.qit);
 
-                    if (_fullScan.AcquisitionMethod == FullScanAcquisitionMethod.DIA &&
-                        _fullScan.IsolationScheme.IsAllIons)
+                    if (_fullScan.IsAllIons)
                     {
                         if (instrumentInfo != null)
                         {
                             _isWatersMse = _isWatersFile;
                             _isAgilentMse = instrumentInfo.IsAgilentFile;
-                            _isElectronIonizationMse = instrumentInfo.ConfigInfoList != null &&
-                                   instrumentInfo.ConfigInfoList.Any(c => @"electron ionization".Equals(c.Ionization));
+                            _isElectronIonizationMse = (instrumentInfo.ConfigInfoList != null &&
+                                                        instrumentInfo.ConfigInfoList.Any(c => @"electron ionization".Equals(c.Ionization))) ||
+                                                       _acquisitionMethod == FullScanAcquisitionMethod.EI;
```

Comment:
I wonder whether this code should be moved outside of the `if (instrumentInfo != null)` block, since it does not actually require an instrumentInfo to check the acquisition method.
Maybe this statement should be:
```
_isElectronIonizationMse = 
    true == instrumentInfo?.ConfigInfoList?.Any(c => @"electron ionization".Equals(c.Ionization))
    || _acquisitionMethod == FullScanAcquisitionMethod.EI;
```

Reply:
Cleaner, yes - thanks